src/core/Akka/Event/LogFilter.cs

```diff
@@ -0,0 +1,319 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="LogFilter.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2024 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2024 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Akka.Actor;
+using Akka.Actor.Setup;
+
+namespace Akka.Event;
+
+/*
+ * NOTE: We do not do caching here - the number of log sources can be very large and
```

Comment:
**Do not do caching here** - no real utility since we're out of band anyway, and the variety of logs is utterly massive. You want better performance? DON'T EMIT THE LOGS IN THE FIRST PLACE.